File: client/src/app/auth/random-password.js
(function() {
  'use strict';

  var app = angular.module('radar.auth');

  app.factory('randomPassword', ['adapter', function(adapter) {
    return function randomPassword() {
      return adapter.get('/random-password').then(function(response) {
        return response.data.password;
      });
    };
  }]);
})();

File: api/radar_api/views/random_password.py
from radar.auth.passwords import generate_password
from radar.serializers.core import Serializer
from radar.serializers.fields import StringField
from radar.views.core import ApiView, response_json


class PasswordSerializer(Serializer):
    password = StringField()


class RandomPasswordView(ApiView):
    @response_json(PasswordSerializer)
    def get(self):
        password = generate_password()
        return {'password': password}


def register_views(app):
    app.add_public_endpoint('random_password')
    app.add_url_rule('/random-password', view_func=RandomPasswordView.as_view('random_password'))
